jobfunnel/tools/easy_apply.py

```diff
@@ -0,0 +1,125 @@
+import csv
+import subprocess
+import random
+import webbrowser
+import sys
+import argparse
+import os
+
+
+def extract_link(row):
+    """
+    extracts the application link from a csv row.
+    """
+    return row[len(row)-4]
+
+
+def duplicate_csv(new_file, old_file):
+    """
+    This takes the contents of old_file and dumps them onto new_file.
+    It is assumed that both files are already open.
+    I recommend openning the file with mode='r+' to make your life easy.
+    """
+    this_new_file = csv.writer(new_file, delimiter=',')
+    this_old_file = csv.reader(old_file, delimiter=',')
+    for row in this_old_file:
+        this_new_file.writerow(row)
+    new_file.seek(0)
+    old_file.seek(0)
+
+
+def count_csv_rows(csvfile):
+    """
+    Counts all of the rows in csvfile.
+    csvfile is assumed to be already open.
+    """
+    csvfile.seek(0)
+    csvreader = csv.reader(csvfile, delimiter=',')
+    count = 0
+    for row in csvreader:
+        count += 1
+    csvfile.seek(0)
+    print("count=", count)
+    return count
+
+def get_not_new_list(csvfile):
+    csvfile.seek(0)
+    csvreader = csv.reader(csvfile, delimiter=',')
+    not_new = []
+    count = 0
+    for row in csvreader:
+        if row[0] != "new":
+            not_new.append(count)
+        count += 1
+    csvfile.seek(0)
+    return not_new
+
+def easy_apply(output_file, input_file, status, number_of_times):
+    """
+    This modeule integrates and uses JobFunnel. It is especifically designed to work with JobFunnel.
+    It will randomly open up as many tabs as you tell it to on your browser from your master_list file that
+    is generated by JobFunnel. It will also mark each one of the randomly selected rows with a status you give it.
+    The default status is 'archive'. After running the easy_apply function, you should see a "updated_master_list.csv" file
+    on the same directory you ran the script on. And your master_list.csv file should have some updated rows with
+    the status you passed.
+    Hope it helps you on your job hunting as much as it has helped me :). 
+    """
+    print("running jobfunnel_easy_apply:",[output_file,input_file,status,number_of_times])
```

Comment:
Super helpful method - does it serve the iser jobs which have already been archived or rejected?

Reply:
As of the moment, if something is already "archive" then it just ignores those jobs. The only jobs that are not ignored by easy_apply are those with the status of "new". Do you think this is the best approach, or should I do something about a job that is already "archive"? My thinking was that if something is "archive", then JobFunnel will ignore those jobs. Maybe I misread the docs, but doesn't JobFunnel remove the "archive" jobs? The workflow I had in mind with easy_apply was the following:

1. Scrape jobs with JobFunnel
2. Go get coffee/tea while you wait
3.Use easy_apply to apply for jobs
4. Once you're done applying to all jobs(at this point in my intended use case 100% of jobs are "archive"). 
5. Got back to step 1.